src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -226,6 +227,39 @@ void ProtocolHandlerImpl::SendStartSessionAck(
   bson_object_deinitialize(&empty_param);
 }
 
+void ProtocolHandlerImpl::WriteProtocolVehicleData(
+    BsonObject& params, const connection_handler::ProtocolVehicleData& data) {
+  auto write_string_to_bson = [&params](const std::string& param_name,
+                                        const std::string& param_value) {
+    if (param_value.empty()) {
+      return;
+    }
+
+    const uint16_t max_string_length = 500;
+    char value_buffer[max_string_length + 1];  // extra byte for NULL symbol
+    strncpy(value_buffer, param_value.c_str(), sizeof(value_buffer));
+    value_buffer[max_string_length] = 0;
```

Comment:
@iCollin we have to do that trick just in order to convert `std::string` to `char*` because bson library accepts only `char*` argument. Note that `c_str()` is not applicable here.
Please let me know if you have any ideas how to make this code more simpler

Reply:
That is a good reason. I might use `const_cast` in this case, but it is fine as is.

Reply:
@iCollin agree. Const cast might cause an undefined behavior if by some reason 3rd party library modifies the immutable variable, that's why I won't use this here :)